drivers/ethernet/phy/phy_lan8720a.c

```diff
@@ -0,0 +1,449 @@
+	struct phy_link_state new_state;
+	int ret;
+	uint32_t irq_value;
+
+	ret = phy_lan8720a_read(data->self, LAN8720A_INTERRUPT_SOURCE_REG, &irq_value);
+	if (ret < 0) {
+		LOG_ERR("Failed to read interrupt register (%d)", ret);
+		return;
+	}
+
+	ret = phy_lan8720a_get_link(data->self, &new_state);
+	if (ret < 0) {
+		return;
+	}
+
+	if (memcmp(&data->state, &new_state, sizeof(struct phy_link_state)) != 0) {
+		memcpy(&data->state, &new_state, sizeof(struct phy_link_state));
+
+		if (data->cb) {
+			data->cb(data->self, &data->state, data->cb_data);
+		}
+	}
+}
+
+static int phy_lan8720a_init(const struct device *dev)
+{
+	const struct lan8720a_config *cfg = dev->config;
+	struct lan8720a_data *data = dev->data;
+	int ret;
+
+	data->self = dev;
+
+	(void)k_mutex_init(&data->mutex);
+	k_work_init(&data->irq_work, phy_lan8720a_work_handler);
+
+#if LAN8720A_RST_GPIO_SUPPORT
+	if (cfg->rst_gpio.port != NULL) {
+		if (!gpio_is_ready_dt(&cfg->rst_gpio)) {
+			LOG_ERR("RST pin not ready");
+			return -ENODEV;
+		}
+
+		/* Start reset */
+		ret = gpio_pin_configure_dt(&cfg->rst_gpio, GPIO_OUTPUT_ACTIVE);
+		if (ret < 0) {
+			LOG_ERR("Failed to configure RST pin (%d)", ret);
+			return ret;
+		}
+
+		/* 100us assertion time */
+		k_busy_wait(100);
+
+		ret = gpio_pin_set_dt(&cfg->rst_gpio, 0);
+		if (ret < 0) {
+			LOG_ERR("Failed to de-assert RST pin (%d)", ret);
+			return ret;
+		}
+	}
+#endif
+
+	if (gpio_is_ready_dt(&cfg->irq_gpio)) {
+		ret = gpio_pin_configure_dt(&cfg->irq_gpio, GPIO_INPUT);
+		if (ret < 0) {
+			LOG_ERR("Failed to configure INT pin (%d)", ret);
+			return ret;
+		}
+
+		ret = gpio_pin_interrupt_configure_dt(&cfg->irq_gpio, GPIO_INT_EDGE_TO_ACTIVE);
+		if (ret < 0) {
+			LOG_ERR("Failed to configure INT (%d)", ret);
+			return ret;
+		}
+
+		gpio_init_callback(&data->irq_callback, phy_lan8720a_isr, cfg->irq_gpio.pin);
+		ret = gpio_add_callback_dt(&cfg->irq_gpio, &data->irq_callback);
+		if (ret < 0) {
+			LOG_ERR("Failed to add INT callback (%d)", ret);
+			return ret;
+		}
+	} else {
+		LOG_ERR("IRQ pin not ready");
+		return -ENODEV;
+	}
+
+	mdio_bus_enable(cfg->mdio_dev);
+
+	return 0;
+}
+
+static const struct ethphy_driver_api lan8720a_phy_api = {
+	.soft_reset = phy_lan8720a_soft_reset,
+	.get_link = phy_lan8720a_get_link,
+	.cfg_link = phy_lan8720a_cfg_link,
+	.link_cb_set = phy_lan8720a_link_cb_set,
+	.read = phy_lan8720a_read,
+	.write = phy_lan8720a_write,
+};
+
+#if LAN8720A_RST_GPIO_SUPPORT
+#define LAN8720A_RST_GPIO_INIT(inst)							\
+	.rst_gpio = GPIO_DT_SPEC_INST_GET_OR(inst, reset_gpios, {0}),
+#else /* LAN8720A_RST_GPIO_SUPPORT */
+#define LAN8720A_RST_GPIO_INIT(inst)
+#endif /* !LAN8720A_RST_GPIO_SUPPORT */
+
+#define LAN8720A_INIT(inst)								\
+	static const struct lan8720a_config lan8720a_##inst##_config = {		\
+		.addr = DT_INST_REG_ADDR(inst),						\
+		.mdio_dev = DEVICE_DT_GET(DT_INST_PARENT(inst)),			\
+		LAN8720A_RST_GPIO_INIT(inst)						\
```

Comment:
I guess something like that should work:

<pre>
   IF_ENABLED(DT_INST_NODE_HAS_PROP(reset_gpios),                                  \
                        (.rst_gpio = GPIO_DT_SPEC_INST_GET(inst, reset_gpios),))   \
</pre>

Reply:
i tried that, didn't compile.

Reply:
Weird, it's commonly used. Would you mind pasting the errors here?

Reply:
Was using the wrong macro, fixed now.